src/core/connection.c

```diff
@@ -5626,6 +5656,34 @@ QuicConnProcessUdpUnreachable(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicConnProcessRouteCompletion(
+    _In_ QUIC_CONNECTION* Connection,
+    _In_ const uint8_t* PhysicalAddress,
+    _In_ BOOLEAN Succeeded
+    )
+{
+    if (Succeeded) {
+        QUIC_PATH* Path = &Connection->Paths[0];
```

Comment:
This code is too fragile to assume just `Paths[0]`. What if there are multiple parallel paths?